### PR TITLE
feat: support managing cluster users

### PR DESCRIPTION
**Pull request by mikechesterwang**
Note that the control plane does not support getting password. 

The password will be stored in the state in `Create` and `Update` operations to avoid inconsistency between plan and state. 
But the `Read` operation will try to read password from the state since we do not know the password at the cloud side.

### CODE REVIEW

**Comment by mikechesterwang** [comment]
The acceptance test is passed.

**Comment by wjf3121** [comment]
> So the update password API will be called and the plan show there is a diff in the password field everytime terraform plan is called.

It seems that the common practice is to store the password in the TF state file and compare it with the value in the config. TF state file is generally considered confidential

**Comment by mikechesterwang** [comment]
> > So the update password API will be called and the plan show there is a diff in the password field everytime terraform plan is called.
> 
> It seems that the common practice is to store the password in the TF state file and compare it with the value in the config. (Both RDS and cloud SQL do that).

Okay, I just made it to read the password from the state.
But, in this case, if user change the password somewhere else, Terraform won't be aware of that. But I guess it is fine since it is the common practice.

File: internal/provider/resource_cluster_user.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+	}
+}
+
+func (r *ClusterUserResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	nsID, username := parseClusterUserIdentifier(req.ID, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if _, err := r.client.GetClusterUser(ctx, nsID, username); err != nil {
```
Comment (wjf3121) [inline internal/provider/resource_cluster_user.go:311]: To confirm the behavior: password won't be populated after the import and the next apply will update the password to the one specified in the config, right?

**Review by wjf3121 — COMMENTED** [review]
Looks good! I will take a deeper look at the implementation tomorrow.

File: internal/provider/acc/provider_test.go
[the remark this answers is not in the document]
Comment (mikechesterwang) [reply internal/provider/acc/provider_test.go:1]: I plan to seperate it from unit tests

File: internal/provider/resource_cluster_user.go (hunk 1 above)
Comment (mikechesterwang) [reply internal/provider/resource_cluster_user.go:311]: correct.

File: docs/index.md [hunk 2]
```diff
@@ -8,7 +8,6 @@ description: |-
 
 # risingwavecloud Provider
 
-**This project is under heavy development. Please join our [Slack](https://join.slack.com/t/risingwave-community/shared_invite/zt-1jei7dk79-fguGadPI2KnhtWnnxBVGoA) to get the latest information.**
```
[the remark this answers is not in the document]
Comment (mikechesterwang) [reply docs/index.md:11]: need to figure out how to keep the custom content in docs generation.

File: internal/provider/acc/provider_test.go
[the remark this answers is not in the document]
Comment (wjf3121) [reply internal/provider/acc/provider_test.go:1]: Oh, does acc mean acceptance test? I thought it was account. Will calling it `acctest` be a bit more clear?

**Review by wjf3121 — APPROVED** [review]
Looks good other than the unresolved comments!

File: internal/provider/acc/provider_test.go
[the remark this answers is not in the document]
Comment (mikechesterwang) [reply internal/provider/acc/provider_test.go:1]: sure, I'll rename it in another PR

**Review by wjf3121 — APPROVED** [review]

**Pull request merged**